src/main/scala/sbtbuildinfo/ScalaCaseObjectRenderer.scala

```diff
@@ -63,16 +63,20 @@ private[sbtbuildinfo] case class ScalaCaseObjectRenderer(options: Seq[BuildInfoO
   def toJsonLine: Seq[String] =
     if (options contains BuildInfoOption.ToJson)
       List(
-         """  val toJson: String = toMap.map{ i =>
-           |    def quote(x:Any) : String = "\"" + x + "\""
-           |    val key : String = quote(i._1)
-           |    val value : String = i._2 match {
-           |       case elem : Seq[_] => elem.map(quote).mkString("[", ",", "]")
-           |       case elem : Option[_] => elem.map(quote).getOrElse("null")
-           |       case elem => quote(elem)
+         """  private def quote(x: Any): String = "\"" + x + "\""
+           |
+           |  private def toJsonValue(value: Any): String = {
+           |    value match {
+           |      case elem: Seq[_] => elem.map(toJsonValue).mkString("[", ",", "]")
+           |      case elem: Option[_] => elem.map(toJsonValue).orNull
```

Comment:
I don't think we should return `null`. I think returning `"null"` was correct. WDYT?

Reply:
I don't agree, we try to convert to correct json type and `null` is not `"null"`.
For example, we parse `BuildInfo.toJson` and instead of getting for some value `Option[String]` we get `String`.

https://www.json.org/
"A value can be a string in double quotes, or a number, or true or false or **null**, or an object or an array."

Reply:
I see what you mean. We are converting to the correct JSON type, but the string representation of that type.

In that case string representation of JSON null is `"null"`, no?

Reply:
At final we get something like that
```json
{
  "build_number": "1.0-SNAPSHOT",
  "git": {
    "commit": null,
    "commit_date": "2018-05-15T15:48:43+0300",
    "tags": [
      
    ],
    "commit_branch": "develop"
  }
}
```

Reply:
No, I get `null` on line 71 where we work with Option.

Reply:
Correct, that's the code the produces the `null` from the option and gets returned by the call to `toJsonValue`.

The code calling that `toJsonValue` is the code at line 73, which is consuming the map you described in the PR description: https://github.com/sbt/sbt-buildinfo/pull/126#issue-190229646

It's only because your passing it to `String#+` that it works.

Reply:
Sorry, maybe I missed something. Can you explain by example, why you want to return "quoted null"

Reply:
I want to return the string representation of JSON null.

Here, these examples might help:

* `toJsonValue(Some(123))` should return `"123"`
* `toJsonValue(Some("abc"))` should return `"\"abc\""` (quoted string)
* `toJsonValue(None)` should return `"null"`

Another addition we could do is make `toJsonValue(null)` return `"null"`.

Sorry, the fact that the code is recursive and there's multiple levels of strings makes this hard to communicate. 😄